tests/behat/features/bootstrap/LibContext.php

```diff
@@ -0,0 +1,1399 @@
+        case 'lastsearchstring':
+          // Regardless of the choice.
+          $returnString = $this->lastSearchString;
+          break;
+
+        default:
+          throw new Exception("Unknown \$modifier:source combination: " . $string);
+          break;
+
+      }
+    }
+    if ($returnString != $string) {
+      // We always want to tell this, otherwise the tester cannot figure out what was done.
+      print_r("Replaced " . $string . " with " . $returnString);
+      print_r("\n");
+    }
+    return $returnString;
+  }
+
+  /**
+   * Translate popular name to css field name
+   *
+   * @param string $field
+   *   Popular name.
+   *
+   * @return string
+   *   Translated to css field name from popular name. If unknown, input is returned.
+   */
+  private function translateFieldName($field) {
+    $result = $field;
+    switch (strtolower($field)) {
+      case "søg":
+      case "søgefelt":
+        $result = "input#edit-search-block-form--2";
+        break;
+
+    }
+    return $result;
+  }
+
+  /**
+   * Runs through the facets and deselects one. Note, this will fail if facets have not been selected already
+   *
+   * @When I deselect a facet to increase the search results
+   */
+  public function useFacetsToIncreaseSearchResults() {
+    // Start by logging what we start out with.
+    print_r("Current number of results: " . $this->searchPage->getShownSizeOfSearchResult() . "\n");
+
+    $this->check($this->searchPage->useFacetsToIncreaseSearchResults(), $this->searchPage->getMessages());
+  }
+
+  /**
+   * Runs through the facets and selects the highest number of results possible
+   *
+   * @When I use facets to reduce the search results to the highest possible
+   */
+  public function useFacetsToReduceSearchResultsToTheHighestPossible() {
+    // Start by initialising the stack if necessary and reveal which number we are starting with.
+    if ($this->searchPage->getExpectedSearchResultSize() < 0) {
+      $this->searchPage->setExpectedSearchResultSize($this->searchPage->getShownSizeOfSearchResult());
+    }
+    print_r("Current number of results: " . $this->searchPage->getShownSizeOfSearchResult() . "\n");
+
+    $this->check($this->searchPage->useFacetsToReduceSearchResultsToTheHighestPossible(), $this->searchPage->getMessages());
+  }
+
+  /**
+   * Step
+   *
+   * @param int $toPage
+   *   is expected to be numeric. First page is 1.
+   *
+   * @When I use pagination to go to page :toPage
+   */
+  public function usePaginationToGoToPageN($toPage) {
+    // Start by scrolling to the footer so if we fail the screendump will tell us something.
+    $this->searchPage->scrollToBottom();
+
+    // This will return the page number.
+    $curpg = $this->searchPage->getCurrentPage();
+
+    // Only change page if we are not already on it.
+    if ($curpg != $toPage) {
+      $this->check($this->searchPage->goToPage($toPage));
+    }
+    else {
+      // We will not fail this.. it may be on purpose.
```

Comment:
WTF? Explain or remove please.

Reply:
Removed.

Reply:
It was a reminiscence from an early version where this was checked. The comment was a reminder not to fail it. That entire scenario is changed radically since so I should have removed the empty else-clause.